test/exchange_adapters/bitmart.test.ts

```diff
@@ -0,0 +1,190 @@
+
+  const validMockStatusJson = {
+    message: 'OK',
+    code: 1000,
+    trace: '48cff315816f4e1aa26ca72cccb46051.69.16892383896653019',
+    data: {
+      symbols: [
+        {
+          symbol: 'SOLAR_USDT',
+          symbol_id: 2342,
+          base_currency: 'SOLAR',
+          quote_currency: 'USDT',
+          quote_increment: '1',
+          base_min_size: '1.000000000000000000000000000000',
+          price_min_precision: 3,
+          price_max_precision: 6,
+          expiration: 'NA',
+          min_buy_amount: '5.000000000000000000000000000000',
+          min_sell_amount: '5.000000000000000000000000000000',
+          trade_status: '',
+        },
+        {
+          symbol: 'EUROC_USDC',
+          symbol_id: 2632,
+          base_currency: 'EUROC',
+          quote_currency: 'USDC',
+          quote_increment: '0.1',
+          base_min_size: '0.100000000000000000000000000000',
+          price_min_precision: 5,
+          price_max_precision: 8,
+          expiration: 'NA',
+          min_buy_amount: '5.000000000000000000000000000000',
+          min_sell_amount: '5.000000000000000000000000000000',
+          trade_status: 'trading',
+        },
+      ],
+    },
+  }
+
+  const inValidMockStatusJson = {
+    message: 'OK',
+    code: 1000,
+    trace: '48cff315816f4e1aa26ca72cccb46051.69.16892383896653019',
+    data: {
+      symbols: [
+        {
+          symbol: 'SOLAR_USDT',
+          symbol_id: 2342,
+          base_currency: 'SOLAR',
+          quote_currency: 'USDT',
+          quote_increment: '1',
+          base_min_size: '1.000000000000000000000000000000',
+          price_min_precision: 3,
+          price_max_precision: 6,
+          expiration: 'NA',
+          min_buy_amount: '5.000000000000000000000000000000',
+          min_sell_amount: '5.000000000000000000000000000000',
+          trade_status: 'trading',
+        },
+        {
+          symbol: 'EUROC_XOF',
+          symbol_id: 2632,
+          base_currency: 'EUROC',
+          quote_currency: 'USDC',
+          quote_increment: '0.1',
+          base_min_size: '0.100000000000000000000000000000',
+          price_min_precision: 5,
+          price_max_precision: 8,
+          expiration: 'NA',
+          min_buy_amount: '5.000000000000000000000000000000',
+          min_sell_amount: '5.000000000000000000000000000000',
+          trade_status: '',
+        },
+      ],
+    },
+  }
+
+  describe('isOrderbookLive', () => {
+    it('returns true if trade_status is trading', async () => {
+      jest
+        .spyOn(bitmartAdapter, 'fetchFromApi')
+        .mockReturnValue(Promise.resolve(validMockStatusJson))
+      expect(await bitmartAdapter.isOrderbookLive()).toEqual(true)
+    })
+
+    it('returns false if trade_status is not trading', async () => {
+      jest
+        .spyOn(bitmartAdapter, 'fetchFromApi')
+        .mockReturnValue(Promise.resolve(inValidMockStatusJson))
+      expect(await bitmartAdapter.isOrderbookLive()).toEqual(false)
```

Comment:
There are cases in which this could return false:

1. the json response doesn't include `EUROC_USDC`
2. `trade_status` in the `EUROC_USDC` object is != `trading`

Your mock invalid json is only testing for no 1. It would be good to test the 2nd case as well. You can start from the valid json and then extend that to only violate the condition which you want to test. See for example:
https://github.com/celo-org/celo-oracle/blob/main/test/exchange_adapters/binance.test.ts#L138-L163